openwillis/measures/text/util/characteristics_util.py

```diff
@@ -1055,346 +642,318 @@ def update_summ_df(
+            mattr = get_mattr(u)
+            turn_df.loc[idx, cols] = list(sentiment_dict.values()) + [mattr]
+            
+        except Exception as e:
+            logger.error(f"Error in sentiment analysis: {e}")
+            continue
+            
+    sentiment_dict = sentiment.polarity_scores(full_text)
+    mattr = get_mattr(full_text)
 
-    # Calculate the pause time between
-    # each word and add the results to pause_list
-    if measures["pause"] not in df_diff.columns:
-        df_diff[measures["pause"]] = df_diff[time_index[0]].astype(float) - df_diff[
-            time_index[1]
-        ].astype(float).shift(1)
+    summ_df.loc[0, cols] = list(sentiment_dict.values()) + [mattr]
+    df_list = [word_df, turn_df, summ_df]
+    return df_list
 
-    # word-level analysis
-    word_df = get_pause_feature_word(word_df, df_diff, word_list, phrase_index, measures)
+def calculate_file_feature(json_data, model, speakers):
+    """
+    ------------------------------------------------------------------------------------------------------
 
-    # phrase-level analysis
-    phrase_df = get_pause_feature_phrase(
-        phrase_df, df_diff, phrase_list, phrase_index, turn_index, time_index, measures
-    )
+    Calculate file features based on JSON data.
 
-    # turn-level analysis
-    if len(turn_index) > 0:
-        turn_df = get_pause_feature_turn(
-            turn_df, df_diff, turn_list, turn_index, time_index, measures
-        )
+    Parameters:
+    ...........
+    json_conf: list
+        JSON response object.
 
-    # file-level analysis
-    summ_df = update_summ_df(
-        df_diff, summ_df, full_text, time_index, word_df, phrase_df, turn_df, measures
-    )
+    Returns:
+    ...........
+    tuple: A tuple containing two values - the total file length and the percentage of time spent speaking.
 
-    df_feature = [word_df, phrase_df, turn_df, summ_df]
+    ------------------------------------------------------------------------------------------------------
+    """
+    
+    if model == 'aws':
+        segments = json_data.get('items', [])
+        file_length = max(float(segment.get("end_time", "0")) for segment in segments)
+        
+        if speakers is None:
+            return file_length, np.NaN
 
-    return df_feature
+        speaking_time = sum(float(segment.get("end_time", "0") or "0") - float(segment.get("start_time", "0") or "0")
+                           for segment in segments if segment.get("speaker_label", "") in speakers)
+    else:
+        segments = json_data.get('segments', [])
+        file_length = max(segment.get('end', 0) for segment in segments)
+        
+        if speakers is None:
+            return file_length, np.NaN
+        speaking_time = sum(segment['end'] - segment['start'] for segment in segments if segment.get('speaker', '') in speakers)
 
+    speaking_pct = (speaking_time / file_length) * 100
+    return file_length, speaking_pct
```

Comment:
```suggestion
    return file_length/60, speaking_pct
```

minute level instead of seconds

Reply:
Also for some reason `speaking_pct` comes back different from what I expect by running manually, but I am still unsure why

Reply:
- speaking_pct : it includes only timepoints when a given subject (clinician or participant) is speaking, give it a try, it was working fine for me on dummy data.

Reply:
- file_length/60: it will return the file length in minutes, this column is just for the sanity check, so it doesn't matter if file length is in second or minutes. But as sugested will change it to minutes